source/geometry_model/box.cc

```diff
@@ -324,6 +324,24 @@ namespace aspect
       return position_point;
     }
 
+    template <int dim>
+    void
+    Box<dim>::make_periodicity_constraints(AffineConstraints<double> constraints) const
```

Comment:
you are missing the &
```suggestion
    Box<dim>::make_periodicity_constraints(AffineConstraints<double>& constraints) const
```